File: biopython_rewrite/variant_parser.py
#!/usr/bin/env python3
# Parse simple TSV variants and add consequence tag
import sys
if len(sys.argv)<2:
    print('Usage: variant_parser.py variants.tsv'); sys.exit(1)
with open(sys.argv[1]) as fh:
    for line in fh:
        chr,pos,ref,alt = line.strip().split('\t')[:4]
        cons = 'SNP' if len(ref)==len(alt) else 'INDEL'
        print('\t'.join([chr,pos,ref,alt,cons]))

File: legacy_perl/variant_parser.pl
#!/usr/bin/env perl
# Parse simple VCF-like tab delimited variants and annotate with consequence (toy example)
use strict;
use warnings;
my $in = shift or die "Usage: $0 variants.tsv\n";
open my $fh, '<', $in or die $!;
while(<$fh>){
    chomp; my ($chr,$pos,$ref,$alt) = split /\t/;
    my $cons = (length($ref) == length($alt)) ? 'SNP' : 'INDEL';
    print join('\t', $chr,$pos,$ref,$alt,$cons), "\n";
}
